docs/topics/classes.md

```diff
@@ -1,263 +1,560 @@
+[member functions](functions.md) of a class:
+
+```kotlin
+// Class with a primary constructor declaring properties
+class Person(val name: String, var age: Int) {
+    // Member function accessing class properties
+    fun introduce(): String {
+        return "Hi, I'm $name and I'm $age years old."
+    }
 }
 ```
 
-Kotlin has a concise syntax for declaring properties and initializing them from the primary constructor:
+You can also assign default values to properties in the primary constructor:
 
 ```kotlin
-class Person(val firstName: String, val lastName: String, var age: Int)
+class Person(val name: String = "John", var age: Int = 30) { /*...*/ }
 ```
 
-Such declarations can also include default values of the class properties:
+If no value is passed during [instance creation](#creating-instances-of-classes), the property uses the default value:
 
 ```kotlin
-class Person(val firstName: String, val lastName: String, var isEmployed: Boolean = true)
+// Class with a primary constructor including default values for name and age
+class Person(val name: String = "John", var age: Int = 30)
+
+fun main() {
+    // Creates an instance using default values
+    val person = Person()
+    println("Name: ${person.name}, Age: ${person.age}")
+    // Name: John, Age: 30
+}
 ```
+{kotlin-runnable="true"}
 
-You can use a [trailing comma](coding-conventions.md#trailing-commas) when you declare class properties:
+Use the primary constructor parameters to initialize additional class properties directly in the class body:
 
 ```kotlin
+// Class with a primary constructor including default values for name and age
 class Person(
-    val firstName: String,
+    val name: String = "John",
+    var age: Int = 30
+) {
+    // description property initialized in the class body using the primary constructor parameters
+    val description: String = "Name: $name, Age: $age"
+}
+
+fun main() {
+    // Creates an instance of the Person class
+    val person = Person()
+    // Accesses the description property
+    println(person.description)
+    // Name: John, Age: 30
+}
+```
+{kotlin-runnable="true"}
+
+Additionally, you can use a [trailing comma](coding-conventions.md#trailing-commas) in constructor declarations:
+
+```kotlin
+class Person(
+    val name: String,
     val lastName: String,
-    var age: Int, // trailing comma
+    var age: Int,
 ) { /*...*/ }
 ```
 
-Much like regular properties, properties declared in the primary constructor can be mutable (`var`) or read-only (`val`).
+### Initializer blocks
 
-Plain constructor parameters (that are not properties) are accessible in:
-* The class header.
-* Initialized properties within the class body.
-* Initializer blocks.
+The primary constructor can't contain runnable code. Its purpose is to initialize the class and set its
+properties. If you want to run some code during [instance creation](#creating-instances-of-classes), use _initializer blocks_ inside the class body.
```

Comment:
```suggestion
The purpose of the primary constructor is to initialize the class and set its
properties, and usually you can achieve this without needing to run complex code blocks.
If you do need to do this in a primary constructor during [instance creation](#creating-instances-of-classes), use _initializer blocks_ inside the class body.
```

Reply:
Also, maybe add that quite often `init` blocks are used for data validation, e.g., by calling `require` function?

```
class Person(val age: Int) {
    init { require(age > 0, "age must be positive") }
}
```